src/XCCDF/result.c

```diff
@@ -1067,16 +1077,21 @@ xmlNode *xccdf_target_identifier_to_dom(const struct xccdf_target_identifier *ti
 	}
 }
 
-xmlNode *xccdf_rule_result_to_dom(struct xccdf_rule_result *result, xmlDoc *doc, xmlNode *parent, const struct xccdf_version_info* version_info)
+xmlNode *xccdf_rule_result_to_dom(struct xccdf_rule_result *result, xmlDoc *doc, xmlNode *parent, const struct xccdf_version_info* version_info, struct xccdf_benchmark *benchmark, bool use_stig_rule_id)
 {
 	xmlNs *ns_xccdf = lookup_xccdf_ns(doc, parent, version_info);
 
 	xmlNode *result_node = xmlNewTextChild(parent, ns_xccdf, BAD_CAST "rule-result", NULL);
 
 	/* Handle attributes */
 	const char *idref = xccdf_rule_result_get_idref(result);
-	if (idref)
-		xmlNewProp(result_node, BAD_CAST "idref", BAD_CAST idref);
+	if (benchmark && use_stig_rule_id) {
+		struct xccdf_item *item = xccdf_benchmark_get_member(benchmark, XCCDF_RULE, idref);
+		const char *stig_rule_id = xccdf_rule_get_stig_rule_id(XRULE(item));
```

Comment:
we can query this from idents and references without introducing extra methods and extra fields, that would make the patch way smaller